### PR TITLE
Allow staff to hidetext themselves

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

**Comment by AustinXII** [comment]
Jw, why?

**Comment by Zarel** [comment]
Should probably allow any target, if we're going to do it that way.

File: server/chat-commands.js [hunk 1]
```diff
@@ -2598,7 +2598,7 @@ const commands = {
 		if (!targetUser && !room.log.hasUsername(target)) return this.errorReply(`User ${target} not found or has no roomlogs.`);
 		if (!targetUser && !user.can('lock')) return this.errorReply(`User ${name} not found.`);
 		let userid = toId(this.inputUsername);
-		if (!this.can('mute', targetUser, room)) return;
+		if (!(user === targetUser && user.can('mute', null, room)) && !this.can('mute', targetUser, room)) return;
```
Comment (Zarel) [inline server/chat-commands.js]:
```suggestion
```

Comment (whalemer) [reply server/chat-commands.js]: Isn't this just completely removing the permission check?
do you mean ``if (!this.can('mute', null, room)) return;``?

**Comment by whalemer** [comment]
>Jw, why?

I've seen/experienced a number of situations where staff have accidentally said sensitive information (priroom names, potential security issues, even just things like trivia answers). It's a lot easier when you can just hidetext yourself instead of needing to find a RO or leader.

File: server/chat-commands.js (hunk 1 above)
Comment (Zarel) [reply server/chat-commands.js]: Oh yes, I did mean that.

Comment (Zarel) [reply server/chat-commands.js]:
```suggestion
		if (!this.can('mute', null, room)) return;
```

**Pull request merged**